typings/@chainlink/index.d.ts

```diff
@@ -79,6 +80,7 @@ declare module '@chainlink/types' {
   export type ExecuteWithConfig<C extends Config> = (
     input: AdapterRequest,
     config: C,
+    endpoints?: Endpoint[]
```

Comment:
I don't think endpoints fits in with what this type is trying to be.

Or that DEFAULT_ENDPOINT should be part of Config.

They are kind of their own thing which is the function selector picking between an API's url endpoints.